src/linker/Linker.Steps/BodySubstitutionParser.cs

```diff
@@ -0,0 +1,201 @@
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Xml.XPath;
+using Mono.Cecil;
+
+namespace Mono.Linker.Steps
+{
+	public class BodySubstitutionParser : ProcessLinkerXmlStepBase
```

Comment:
Nit: rename `ProcessLinkerXmlStepBase` - it's not a "Step base" anymore.

Reply:
It actually was still a base step - in the case of descriptor xml, it was called from the pipeline. And even for the embedded XML, parsing relied on the Process (LinkContext context) method from `IStep`.

I factored this out some more so that the shared XML processing logic is no longer an `IStep`. There's a new derived class for the descriptor XML, and ResolveFromXmlStep becomes a wrapper for it like the other XML steps.